skimage/measure/_selfoverlapping_polygon.py

```diff
@@ -0,0 +1,1875 @@
+                                                   ex_end=False))
+                else:
+                    sub_poly.append(_shift_indices(child_2[1], a1, n_vertices,
+                                                   ex_start=False,
+                                                   ex_end=False))
+
+                if len(child_poly):
+                    polys_idx.append(np.concatenate(sub_poly, axis=0))
+
+                sub_immersion, child_poly = _immerse_tree(child_2, visited,
+                                                          n_vertices,
+                                                          polys_idx)
+
+                if len(child_poly):
+                    immersion[root].update(sub_immersion)
+                    polys_idx.append(np.concatenate(child_poly, axis=0))
+
+                sub_poly = []
+        break
+
+    return immersion, sub_poly
+
+
+def _get_root_indices(vert_info):
+    """Find the indices of the vertices that define the root cut used to
+    recurse the polygon subdivision algorithm.
+
+    Parameters
+    ----------
+    vert_info : numpy.ndarray
+        An array containing the coordinates of the polygon vertices and
+        additional information about each vertex.
+
+    Returns
+    -------
+    left_idx : int
+        The positional index of the root left vertex
+    right_idx : int
+        The positional index of the root right vertex
+    """
+    n_vertices = vert_info.shape[0]
+
+    # Non-intersection points:
+    org_ids = np.where(vert_info[:, 2].astype(np.int32) == -1)[0]
+
+    root_vertex = org_ids[np.argmax(vert_info[org_ids, 1])]
+
+    shifted_indices = np.mod(root_vertex + 1 + np.arange(n_vertices),
+                             n_vertices)
+
+    right_idx = np.where(
+        vert_info[shifted_indices, 2].astype(np.int32) == 0)[0][0]
+    right_idx = shifted_indices[right_idx]
+
+    shifted_indices = np.mod(root_vertex + np.arange(n_vertices),
+                             n_vertices)
+
+    left_idx = np.where(
+        vert_info[shifted_indices, 2].astype(np.int32) == 0)[0][-1]
+    left_idx = shifted_indices[left_idx]
+
+    return left_idx, right_idx
+
+
+def _get_crest_ids(vertices, tolerance=1e-3):
+    """Find the positional indices of the crest points.
+    Only crests where there is a left turn on the polygon are considered.
+
+    Parameters
+    ----------
+    vertices : numpy.ndarray
+        An array containing the coordinates of the polygon vertices.
+    tolerance : float, optional
+        A tolerance to determine if two vertices are at the same height.
+
+    Returns
+    -------
+    max_crest_ids : list
+        A list of indices of the maximum crest vertices.
+    min_crest_ids : list
+        A list of indices of the minimum crest vertices.
+    max_crest : int
+        The positional index of the maximum crest vertex.
+    min_crest : int
+        The positional index of the minimum crest vertex.
```

Comment:
I think a short explanation / definition of what "crest points" are would be helpful in the long run.

Reply:
Yes, I'll add a definition for what crest points are detected. According to the algorithm, only crest points at _left turns_ are considered. I'll also add what a _left turn_ is for more clarity.